### ISSUE TITLE
Allow a regex for requires in no-var-requires and no-require-imports

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [palantir/tslint#3392]
### Bug Report

- __TSLint version__: 5.7.0
- __TypeScript version__: 2.6
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```typescript
const styles = require("./RootStyles.css");

export const Text = () => <span className={styles.text}>Hello world!</span>;
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "no-require-imports": true,
        "no-var-requires": true
    }
}
```

#### Actual behavior

```
[tslint] require statement not part of an import statement (no-var-requires)
[tslint] require() style import is forbidden (no-require-imports)
```

#### Expected behavior

It's hard to configure CSS modules so that the VS Code TypeScript server understands them. Since they don't give great type definitions out of the box, sometimes the old style var requires are the best option.

Instead of disabling the rules globally, I'd love to have something like what `no-import-side-effect` has:

```json
"no-require-imports": [true, {
    "ignore-module": "(\\.css)$"
}]
```

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
